packages/ui/src/components/Chip/ChipIcon.tsx

```diff
@@ -0,0 +1,84 @@
+import styled from '@emotion/styled'
+import * as Icon from '@ultraviolet/icons'
+import { useContext } from 'react'
+import type { PascalToCamelCaseWithoutSuffix } from 'src/types'
+import { ChipContext } from './ChipContext'
+
+const Container = styled.button`
+  background: none;
+  border:none;
+  cursor: pointer;
+  padding:0;
+  border-radius: ${({ theme }) => theme.radii.default};
+
+  &[data-type="trailing"]{
+    margin-right: -${({ theme }) => theme.space['0.5']};
+  }
+
+  &[data-has-onclick="true"][data-active="false"]:hover{
+      background-color: ${({ theme }) => theme.colors.neutral.backgroundStrongHover};
+  }
+
+  &[data-has-onclick="true"][data-active="true"]:hover{
+      background-color: ${({ theme }) => theme.colors.primary.backgroundStrong};
+  }
+
+  &[data-disabled="true"]{
+    cursor: not-allowed;
+  }
+`
+
+type IconType = PascalToCamelCaseWithoutSuffix<keyof typeof Icon, 'Icon'>
+
+type ChipIconType = {
+  /**
+   * Add an icon in the chip
+   */
+  name: IconType
+  type: 'trailing' | 'leading'
+  onClick?: () => void
+  'data-testid'?: string
+}
+
+export const ChipIcon = ({
+  name,
+  type,
+  onClick,
+  'data-testid': dataTestId,
+}: ChipIconType) => {
+  const { isActive, disabled, chipContext } = useContext(ChipContext)
+
+  if (!chipContext) {
+    throw new Error('Chip.Icon can only be used inside a Chip component.')
+  }
```

Comment:
I think you miss understood what I meant haha.
You don't need to create `chipContext` in a context to check if the context exists. You can simply check directly if the variable returned by `useContext()` actually exists. We have plenty of examples like here: https://github.com/scaleway/ultraviolet/blob/7a47067b94292fa23be258a85a3b48cca7c9cdba/packages/ui/src/components/Modal/ModalProvider.tsx#L27

You could first check the context variable and the destructure it if it exists.